esp/services/ws_cloud/ws_cloudService.cpp

```diff
@@ -59,6 +65,32 @@ bool CWsCloudEx::onGetPODs(IEspContext& context, IEspGetPODsRequest& req, IEspGe
     return true;
 }
 
+const char* CWsCloudEx::readPublicServices(const char* allServices, StringBuffer& publicServices)
+{
```

Comment:
@wangkx you should put a comment that explains that you are following the same logic implemented in kubectl get svc to decide what has an external address and what doesn't.  And then briefly describe the logic found there.